### PR TITLE
Container app may return null in the dapr object, which cli should handle gracefully

### PR DESCRIPTION
**Pull request by calleo**
This fixes https://github.com/Azure/azure-cli-extensions/issues/5781

Needs some guidance on how to add tests for this 🙏 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az containerapp dapr enable

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Container app

**Comment by navba-MSFT** [comment]
@Juliehzl Could you please review this once you get a chance ?

File: src/containerapp/azext_containerapp/custom.py [hunk 1]
```diff
@@ -2366,7 +2366,7 @@ def enable_dapr(cmd, name, resource_group_name,
     if 'configuration' not in containerapp_def['properties']:
         containerapp_def['properties']['configuration'] = {}
 
-    if 'dapr' not in containerapp_def['properties']['configuration']:
+    if not containerapp_def['properties']['configuration'].get('dapr', None):
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/custom.py:2369]: consider using safe_get() to handle such null object in response.

File: src/containerapp/azext_containerapp/custom.py [hunk 2]
```diff
@@ -2366,7 +2366,7 @@ def enable_dapr(cmd, name, resource_group_name,
...
         containerapp_def['properties']['configuration']['dapr'] = {}
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/custom.py:2370]: Add test in https://github.com/Azure/azure-cli-extensions/blob/main/src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py#L583 and follow the pattern in existing test

**Comment by calleo** [comment]
> @calleo please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> 
> Contributor License Agreement

@microsoft-github-policy-service agree [company="{your company}"]

File: src/containerapp/azext_containerapp/custom.py (hunk 1 above)
Comment (calleo) [reply src/containerapp/azext_containerapp/custom.py]: Thank you. Will use that instead 👍

File: src/containerapp/azext_containerapp/custom.py (hunk 2 above)
Comment (calleo) [reply src/containerapp/azext_containerapp/custom.py:2370]: I have added an additional e2e test for this specific scenario. Is there any documentation explaining how to generate the recordings (possibly for a single test). I had to fiddle with [scripts/ci/test_source.py](scripts/ci/test_source.py) to only run the new test method and store the recording. There must be an easier way, but this worked 🙂 

When you get a chance, can you take another look at this PR?

**Comment by Juliehzl** [comment]
@calleo could you help update history.rst with the release note of your fix

**Comment by navba-MSFT** [comment]
> @calleo could you help update history.rst with the release note of your fix

Thanks @Juliehzl .

@calleo I have updated the `history.rst `for your commit. https://github.com/calleo/azure-cli-extensions/pull/1/commits/7243993837f4a3450c12c7e38bbec0e84ec2d21a .

Could you please approve this ?

**Comment by calleo** [comment]
> > @calleo could you help update history.rst with the release note of your fix
> 
> Thanks @Juliehzl .
> 
> @calleo I have updated the `history.rst `for your commit. https://github.com/calleo/azure-cli-extensions/pull/1/commits/7243993837f4a3450c12c7e38bbec0e84ec2d21a .
> 
> Could you please approve this ? 
> 

Thanks! 🙏

**Review by Juliehzl — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**